src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -120,6 +129,16 @@ protected function initializeCatalogue($locale)
         parent::initializeCatalogue($locale);
     }
 
+    protected function doLoadCatalogue($locale): void
```

Comment:
No that won't work, because you'd also need this code triggered when `initializeCacheCatalogue` is called instead of `initializeCatalogue` (i.e., when a `cache_directory` is set).